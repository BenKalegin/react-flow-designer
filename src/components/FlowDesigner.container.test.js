import React from 'react';
import renderer from 'react-test-renderer';
import { Map } from 'immutable';

import { FlowDesigner } from './FlowDesigner.container';
import NodeType from './configuration/NodeType.component';

jest.mock('./ZoomHandler.component');

const noOp = () => {};

describe('<FlowDesigner />', () => {
	it('renders correctly', () => {
		const mockGrid = <g />;
		const nodes = new Map();
		const ports = new Map();
		const links = new Map();
		const tree = renderer
			.create(
				<FlowDesigner
					moveNodeTo={noOp}
					moveNodeToEnd={noOp}
					setNodeTypes={noOp}
					nodes={nodes}
					ports={ports}
					links={links}
					grid={mockGrid}
					reduxMountPoint="mountPoint"
				>
					<NodeType type="test" component={NodeType} />
					<NodeType type="test2" component={NodeType} />
				</FlowDesigner>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('render defs if svgDefs is provided', () => {
		const mockGrid = <g />;
		const nodes = new Map();
		const ports = new Map();
		const links = new Map();
		const svgDefs = (
			<defs>
				<filter id="drop-shadow" width="150%" height="150%">
					<feOffset result="offOut" in="SourceAlpha" dx="0.4" dy="0.4" />
					<feGaussianBlur result="blurOut" in="offOut" stdDeviation="0.1" />
					<feComponentTransfer in="blurOut" result="opacityShadow">
						<feFuncA type="linear" slope="0.2" />
					</feComponentTransfer>
					<feBlend in="SourceGraphic" in2="opacityShadow" mode="normal" />
				</filter>
			</defs>
		);
		const tree = renderer
			.create(
				<FlowDesigner
					svgDefs={svgDefs}
					moveNodeTo={noOp}
					moveNodeToEnd={noOp}
					setNodeTypes={noOp}
					nodes={nodes}
					ports={ports}
					links={links}
					grid={mockGrid}
					reduxMountPoint="mountPoint"
				>
					<NodeType type="test" component={NodeType} />
					<NodeType type="test2" component={NodeType} />
				</FlowDesigner>,
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
