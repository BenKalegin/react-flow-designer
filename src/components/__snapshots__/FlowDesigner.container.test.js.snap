exports[`<FlowDesigner /> render defs if svgDefs is provided 1`] = `
<svg
  onClick={undefined}
  width="100%">
  <defs>
    <filter
      height="150%"
      id="drop-shadow"
      width="150%">
      <feOffset
        dx="0.4"
        dy="0.4"
        in="SourceAlpha"
        result="offOut" />
      <feGaussianBlur
        in="offOut"
        result="blurOut"
        stdDeviation="0.1" />
      <feComponentTransfer
        in="blurOut"
        result="opacityShadow">
        <feFuncA
          slope="0.2"
          type="linear" />
      </feComponentTransfer>
      <feBlend
        in="SourceGraphic"
        in2="opacityShadow"
        mode="normal" />
    </filter>
  </defs>
</svg>
`;

exports[`<FlowDesigner /> renders correctly 1`] = `
<svg
  onClick={undefined}
  width="100%" />
`;
